pkg/secrets/check_rights_nix.go

```diff
@@ -100,15 +101,31 @@ var checkConfigFilePermissions = func(path string) error {
 		return fmt.Errorf("unable to check permissions for '%s': can't stat it: %s", path, err)
 	}
 
-	if stat.Uid != 0 || stat.Gid != 0 {
-		return fmt.Errorf("invalid config file permissions for '%s': not owned by root:root", path)
-	}
-
 	if stat.Mode&syscall.S_IWOTH != 0 {
 		return fmt.Errorf("invalid config file permissions for '%s': cannot have o+w permission", path)
 	}
 
-	return nil
+	usr, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("can't query current user: %s", err)
+	}
+
+	groups, err := usr.GroupIds()
+	if err != nil {
+		return fmt.Errorf("can't query user groups: %s", err)
+	}
+
+	if strconv.FormatInt(int64(stat.Uid), 10) != usr.Uid {
```

Comment:
Is it expected to return an error if the file is owned by the group but not by the user?

Reply:
Yes, I believe so, this is the least ambiguous way - so AND instead of OR.